packages/grafana-data/src/transformations/transformers/formatString.ts

```diff
@@ -35,6 +39,11 @@ const splitToCapitalWords = (input: string) => {
   return arr;
 };
 
+const performRegexReplace = (input: string, regexPattern: string, replacePattern: string) => {
+  const regex = stringToJsRegex(regexPattern);
```

Comment:
@jablonsk19 thanks for the PR :)

i think we should try to avoid re-creating the same regexp for every value. ideally, we would create the regexp once and then only run `replace` on each value.

i had a very similar situation in https://github.com/grafana/grafana/pull/96593 and had to add an additional [getParser()](https://github.com/grafana/grafana/pull/96593/files#diff-601e183dc56c031bed480c34d0a002a06d6646439454811c379aaa153a3d5230R24) layer to put the compiled-once regexp into the replacer function's scope. please take a look and see if a similar approach can be applied here.